### ISSUE TITLE
Ctx.Output.Download returned corrupted files (.xlsx, .zip) in Windows

### ISSUE THREAD

**Issue by har07** [astaxie/beego#2591]
1. `bee version`:

		| ___ \
		| |_/ /  ___   ___
		| ___ \ / _ \ / _ \
		| |_/ /|  __/|  __/
		\____/  \___| \___| v1.8.2

		├── Beego     : 1.7.2
		├── GoVersion : go1.8
		├── GOOS      : windows
		├── GOARCH    : amd64
		├── NumCPU    : 8
		├── GOPATH    : D:\GoWorkspace
		├── GOROOT    : C:\Go\
		├── Compiler  : gc
		└── Date      : Monday, 24 Apr 2017

2. `go env`:

		set GOARCH=amd64
		set GOBIN=
		set GOEXE=.exe
		set GOHOSTARCH=amd64
		set GOHOSTOS=windows
		set GOOS=windows
		set GOPATH=D:\GoWorkspace
		set GORACE=
		set GOROOT=C:\Go
		set GOTOOLDIR=C:\Go\pkg\tool\windows_amd64
		set GCCGO=gccgo
		set CC=gcc
		set GOGCCFLAGS=-m64 -mthreads -fmessage-length=0
		set CXX=g++
		set CGO_ENABLED=1
		set PKG_CONFIG=pkg-config
		set CGO_CFLAGS=-g -O2
		set CGO_CPPFLAGS=
		set CGO_CXXFLAGS=-g -O2
		set CGO_FFLAGS=-g -O2
		set CGO_LDFLAGS=-g -O2

3. What did I do?

- create new api: `bee api beego-test`
- add new controller method and call `` within the method : 

		func (o *ObjectController) Download() {
			o.Ctx.Output.Download("Book1.xlsx", "Book1.xlsx")
		}

- create blank excel file, Book1.xlsx, in the root folder of the project (directly in the beego-test folder to be clear)

4. What did you expect to see?

When I call the API endpoint that invoke `Download()` method, I'd expect the downloaded file to be exactly the same as the Book1.xlsx in the project folder.

5. What did you see instead?

The downloaded file can not be opened by Ms Excel

![2017-04-24 11_38_50-excel product activation failed](https://cloud.githubusercontent.com/assets/11190794/25322635/dc552df2-28e2-11e7-92a4-138b145cdc8f.png)

 I tested using .zip and .json (plain text) files as well, the former showed the same problem (Winrar complained about "Unexpected end of archive") while the latter produced a notepad readable text file (worked fine). I didn't have OS other than Windows to verify whether this issue is specific to Windows or it can be reproduced in Unix/Mac as well.

![image](https://cloud.githubusercontent.com/assets/11190794/25335197/3a544a62-291c-11e7-953f-adeb40c11c0a.png)

**Comment by har07** [comment]
The same issue happened when I tried, instead of using `Ctx.Output.Download()`, manually setting response headers (following [the source](https://github.com/astaxie/beego/blob/master/context/output.go#L250)) and write bytes directly using `o.Ctx.ResponseWriter.Write()`

**Comment by har07** [comment]
And this is sample .zip file I used to reproduce the issue : [temp.zip](https://github.com/astaxie/beego/files/951185/temp.zip)

```
func (o *ObjectController) Test() {
	// o.Ctx.Output.Download("Book1.xlsx", "Book1.xlsx")
	o.Ctx.Output.Download("temp.zip", "temp.zip")
}
```

**Comment by har07** [comment]
Fixed after update to bee 1.8.3 initially suggested by astaxie [in the other issue report](https://github.com/astaxie/beego/issues/2524#event-1054847704). My setup when this issue gone : 

		λ bee version
		______
		| ___ \
		| |_/ /  ___   ___
		| ___ \ / _ \ / _ \
		| |_/ /|  __/|  __/
		\____/  \___| \___| v1.8.3

		├── Beego     : 1.8.0
		├── GoVersion : go1.8
		├── GOOS      : windows
		├── GOARCH    : amd64
		├── NumCPU    : 8
		├── GOPATH    : D:\GoWorkspace
		├── GOROOT    : C:\Go\
		├── Compiler  : gc
		└── Date      : Tuesday, 25 Apr 2017

**Comment by astaxie** [comment]
Thanks for the update
